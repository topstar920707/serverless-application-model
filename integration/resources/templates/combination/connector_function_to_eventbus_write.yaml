Resources:
  LambdaRole:
    Type: AWS::IAM::Role
    Properties:
      AssumeRolePolicyDocument:
        Statement:
          - Effect: Allow
            Action: sts:AssumeRole
            Principal:
              Service: lambda.amazonaws.com
      ManagedPolicyArns:
        - !Sub arn:${AWS::Partition}:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole

  TriggerFunction:
    Type: AWS::Lambda::Function
    Properties:
      Role: !GetAtt LambdaRole.Arn
      Runtime: nodejs14.x
      Handler: index.handler
      Code:
        ZipFile: |
          const AWS = require('aws-sdk');
          var put_params = {
            Entries: [
              {
                EventBusName: process.env.EventBusName,
                Source: "com.mycompany.myapp",
                Detail:"{ \"key1\": \"value1\", \"key2\": \"value2\" }",
                Resources:[
                  "resource1",
                  "resource2"
                ],
                DetailType:"myDetailType"
              }
            ]
          }
          exports.handler = async (event) => {
            console.log('REQUEST RECEIVED:', JSON.stringify(event));
            var eventbridge = new AWS.EventBridge();
            await eventbridge.putEvents(put_params).promise();
          };
      Environment:
        Variables:
          EventBusName: !Ref EventBus

  EventBus:
    Type: AWS::Events::EventBus
    Properties: 
      Name: TestEventBus

  Connector:
    Type: AWS::Serverless::Connector
    Properties:
      Source:
        Id: TriggerFunction
      Destination: 
        Id: EventBus
      Permissions:
        - Write