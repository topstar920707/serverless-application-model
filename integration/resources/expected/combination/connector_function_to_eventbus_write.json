[
  {
    "LogicalResourceId": "LambdaRole",
    "ResourceType": "AWS::IAM::Role"
  },
  {
    "LogicalResourceId": "TriggerFunction",
    "ResourceType": "AWS::Lambda::Function"
  },
  {
    "LogicalResourceId": "EventBus",
    "ResourceType": "AWS::Events::EventBus"
  },
  {
    "LogicalResourceId": "ConnectorPolicy",
    "ResourceType": "AWS::IAM::ManagedPolicy"
  }
]